### ISSUE TITLE
bug: ion-icon on iOS is buggy when using custom SVGs

### ISSUE THREAD

**Issue by eduardoRoth** [ionic-team/ionic-framework#23659]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [ ] v6.x

### Current Behavior

When using a custom SVG with `ion-icon` on iOS I get this weirdly colored SVG. This doesn't happen on Android. This happens on emulator devices and physical ones too.

Sometimes it flickers between the right SVG and the wrongly colored one:

Usually it looks like this:
![image](https://user-images.githubusercontent.com/5419161/126360380-fed0dde1-73e2-48d6-a363-cef7895da8e4.png)

But if I navigate in the app then the SVG looks fine
![image](https://user-images.githubusercontent.com/5419161/126360397-852caab2-5e06-4dcb-8a80-17d25a03f609.png)

But if I open an `ion-menu` then the SVG looks like this
![image](https://user-images.githubusercontent.com/5419161/126360498-29df6606-b525-47de-b085-17772d8707c7.png)

### Expected Behavior

`ion-icon` SVG should be shown correctly, as on Android

### Steps to Reproduce

Create an `ion-icon` tag inside an `ion-item` `slot=start`, use `src` attribute to point to an SVG in you `assets` folder and run on iOS.

### Ionic Info

```
Ionic:

   Ionic CLI                     : 6.16.3
   Ionic Framework               : @ionic/angular 5.6.11
   @angular-devkit/build-angular : 12.1.2
   @angular-devkit/schematics    : 8.3.29 
   @angular/cli                  : 12.1.2
   @ionic/angular-toolkit        : 4.0.0

Capacitor:

   Capacitor CLI      : 3.1.1
   @capacitor/android : 3.1.1
   @capacitor/core    : 3.1.1
   @capacitor/ios     : 3.1.1

Utility:

   cordova-res : 0.15.3
   native-run  : 1.4.0

System:

   NodeJS : v14.17.3
   npm    : 6.14.13
   OS     : macOS Big Sur

```

**Comment by eduardoRoth** [comment]
Should I prefer using an `ion-thumbnail` element or an image tag?

**Comment by eduardoRoth** [comment]
Ok, seems to be partially fixed, had this:

```
<ion-icon slot="start" [src]="iconoUrl" *ngIf="iconoUrl"></ion-icon>
```

And changed it to this:

```
<ng-container *ngIf="iconoUrl">
    <ion-icon slot="start" [src]="iconoUrl"></ion-icon>
  </ng-container>
```

SVG now renders fine but when I open the menu and close it from the backdrop, the icon looks like this:

![image](https://user-images.githubusercontent.com/5419161/126375960-d0bebf54-fca2-4994-bfbb-47d1a64b143d.png)

**Comment by eduardoRoth** [comment]
Using the following works with no problem (although I'd need to change the size of the img):

```
<ion-thumbnail slot="start" *ngIf="iconoUrl">
    <img [src]="iconoUrl" />
</ion-thumbnail>
```

![image](https://user-images.githubusercontent.com/5419161/126377518-bb46c0ba-0e4b-4d44-a1a4-07d993604fb5.png)

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you reproduce the issue in an Ionic starter app and provide a link to the repo? I am not seeing this on my end.

**Comment by Lxstr** [comment]
I noticed this happening in my app when using colored svgs and within any parent elements that have a color attribute ie color="primary". For me it was also intermittent and sometimes only showed when the app was left and re-entered.

I suspected this was due to the inheritance of the color attributes overriding the colors in the svg file.

My solution at the time was to only use colorless icons and specifically apply the color to the icon element. At some point I was planning to revisit it though and look deeper.
